pkgs/top-level/all-packages.nix

```diff
@@ -22068,6 +22068,57 @@ with pkgs;
                 );
     }).config.system.build;
 
+
+  /*
+   * Run a NixOS VM network test.
+   *
+   * This is equivalent to  import ./make-test.nix  from the NixOS manual
+   * For details see https://nixos.org/nixos/manual/index.html#sec-nixos-tests
+   *
+   * Parameter:
+   *   A NixOS VM test network, or path to it. Example:
+   *
+   *      { lib, ... }:
+   *      { name = "my-test";
+   *        nodes = {
+   *          machine-1 = someNixOSConfiguration;
+   *          machine-2 = ...;
+   *        }
+   *      }
+   *
+   * Result:
+   *   A derivation that runs the VM test.
+   *
+   * For the interaction between Nixpkgs and NixOS configuration
+   * consult the pkgs.nixos function documentation.
+   */
+  nixosTest =
+    let
+      /* The nixos/lib/testing.nix module, preapplied with arguments that
+       * make sense for this evaluation of Nixpkgs.
+       */
+      nixosTesting =
+        (import ../../nixos/lib/testing.nix {
+          inherit (pkgs.stdenv.hostPlatform) system;
+          inherit pkgs;
+          extraConfigurations = [(
+            { lib, ... }: {
+              config.nixpkgs.pkgs = lib.mkDefault pkgs;
+            }
+          )];
+        });
+    in
+      test:
+        let
+          loadedTest = if builtins.typeOf test == "path"
+                       then import test
+                       else test;
+          calledTest = if pkgs.lib.isFunction loadedTest
+                       then callPackage loadedTest {}
+                       else loadedTest;
+        in
+          nixosTesting.makeTest calledTest;
```

Comment:
Hm, the function only accepts one test. I don’t know how much sharing is done internally, this might mean a hefty evaluation for each test if you want to run multiple tests. cc @aszlig maybe.

Reply:
This reuses an applied Nixpkgs (`pkgs`), so that will be efficient. Evaluation nixos configurations is inevitable because each machine in a nixos test is unique.